library/src/main/java/com/timehop/stickyheadersrecyclerview/util/OrientationProvider.java

```diff
@@ -0,0 +1,12 @@
+package com.timehop.stickyheadersrecyclerview.util;
+
+import android.support.v7.widget.RecyclerView;
+
+/**
+ * Interface for getting the orientation of a RecyclerView from its LayoutManager
+ */
+public interface OrientationProvider {
```

Comment:
Ideally this library wouldn't necessarily rely on the LinearLayout manager for orientation cues, which is the specific provider I implemented.